### PR TITLE
ci: auto-publish `master` commits to npm

### PR DESCRIPTION
**Pull request by stephenmathieson**
This patch updates our CircleCI configuration, enabling it to automatically publish green `master` commits to npm.

This added configuration does not attempt to generate a CHANGELOG entry, update `package.json#version`, or make any other metadata change. It is expected that the developer creating the release will do this.

For more information, see the [whitepaper in Google Drive](https://docs.google.com/document/d/14S2ktHpBVNPAR7JdkptOOgmufbZIV0m7orz6N95tPzk/edit) (private paper, sorry OSS folks).

### CODE REVIEW

**Comment by stephenmathieson** [comment]
@WilcoFiers I do not have access to this repository's settings page, so I'm unable to update the required status check configuration. This needs to happen before this can land.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
+version: 2
+
+defaults: &defaults
+  docker:
+    - image: circleci/node:6.12.3-browsers
+  working_directory: ~/axe-core
```
Comment (WilcoFiers) [inline .circleci/config.yml]: Should this be `~/axe-core` @stephenmathieson?

File: package.json [hunk 2]
```diff
@@ -24,6 +24,7 @@
   },
   "devDependencies": {
     "grunt": "~0.4.5",
+    "grunt-cli": "^1.3.1",
```
Comment (WilcoFiers) [inline package.json:27]: What's this for?

File: .circleci/config.yml (hunk 1 above)
Comment (stephenmathieson) [reply .circleci/config.yml]: Nope, good catch. Copypasta fail.

File: package.json (hunk 2 above)
Comment (stephenmathieson) [reply package.json:27]: It's for using Grunt. This project doesn't work "as-is" and needed this dependency added. CI was failing until https://github.com/dequelabs/grunt-axe-webdriver/pull/40/commits/f9cc4146b5905f8e267790498c6c722aac340197.

Comment (jeeyyy) [reply package.json:27]: CI still fails, so unsure if above fixes it.

Comment (stephenmathieson) [reply package.json:27]: Adding this dependency is 100% necessary. The `grunt` program won't exist otherwise.

CI failed due to a caching issue.

**Review by jeeyyy — APPROVED** [review]
👍

**Pull request merged**